Neos.ContentRepositoryRegistry/Classes/Service/EventMigrationService.php

```diff
@@ -298,7 +298,9 @@ public function migrateMetaDataToWorkspaceName(\Closure $outputFn): void
 
             // ... and update the event
             // the payload is only used for rebasing where we override the workspace either way:
-            $eventMetaData['commandPayload']['workspaceName'] = 'missing:' . ($eventMetaData['commandPayload']['contentStreamId'] ?? '');
+            $eventMetaData['commandPayload']['workspaceName'] = WorkspaceName::fromString(
+                'missing-' . \mb_substr($eventMetaData['commandPayload']['contentStreamId'] ?? '', 0, 28)
+            )->value;
```

Comment:
Good catch, but with that we'll expose internals. i.e. if the max WorkspaceName length were about to change (even if unlikely) this could break again.
I would suggest to either use [WorkspaceName::MAX_LENGTH](https://github.com/neos/neos-development-collection/blob/2cdeb6a72d2b08ef75f1d8096bc5f679734e8a57/Neos.ContentRepository.Core/Classes/SharedModel/Workspace/WorkspaceName.php#L26) or (probably better) a combination with the dedicated transliteration constructor:
```suggestion
            $eventMetaData['commandPayload']['workspaceName'] = WorkspaceName::transliterateFromString(
                'missing-' . \mb_substr($eventMetaData['commandPayload']['contentStreamId'] ?? '', 0, WorkspaceName::MAX_LENGTH - 8)
            )->value;
```

Reply:
Agreed, using the constant makes more sense than some arbitrary integer

Reply:
you kept `fromString` instead of `transliterateFromString` – intentionally?